### PR TITLE
Use abvm/blwm on USE scripts and Khmer

### LINKED ISSUE #566: "Indic" glyph separation breaks mkmk
In the mark feature writer, marks belonging to Indic glyphs are placed in abvm/blwm features and other marks are positioned using mark/mkmk.

However... the Sharada script (which goes through the USE and is not considered "Indic" by ufo2ft) uses the Devanagari udatta (U+0951). When a Sharada font is compiled with ufo2ft, all the Sharada mark positioning rules go into mark/mkmk and the "Indic" rules (containing udatta) go into abvm. The glyphs are separated out into Indic/non-Indic before creating these rules, and this means there is no interaction between Sharada marks and Devanagari marks; i.e. there is no way to place a udatta on a Sharada mark such as a Candrabindu.

I don't know if that's linguistically something that you might want to do, but e.g. Noto Sans Sharada allows it, positioning all marks in the mkmk feature.

I don't know how best to fix this:

* Add USE scripts to the `indicScripts` list?
* Provide a way of opting out of the `abvm`/`blwm` writer?

### PR DESCRIPTION
**Pull request by simoncozens**
- Add constants for list of Indic/USE scripts
- Add USE and Khmer to abvm/blwm scripts

Fixes #566, #565, #522

### CODE REVIEW

**Review by anthrotype — APPROVED** [review]
LGTM!

File: Lib/ufo2ft/constants.py [hunk 1]
```diff
@@ -37,3 +37,95 @@
...
+    "Orya",  # Oriya
+    "Taml",  # Tamil
+    "Telu",  # Telugu
+]
+
+USE_SCRIPTS = [
+    # Correct as at Unicode 14.0
+    "Tibt",  # Tibetan
+    "Mong",  # Mongolian
+    # HB has Sinhala commented out here?!
```
Comment (behdad) [inline Lib/ufo2ft/constants.py:58]: That's because we send it to Indic. Uncomment.

**Comment by behdad** [comment]
Make sure you handle Myanmar as well.

**Comment by simoncozens** [comment]
MS suggests [using mark for Myanmar](https://docs.microsoft.com/en-us/typography/script-development/myanmar#apply-opentype-gpos-features), not abvm/blwm.

**Comment by khaledhosny** [comment]
With this PR, my kashida mark to base is moved from `mark` feature to `abvm`/`blwm` and they don’t get applied for Arabic except in HarfBuzz.

**Comment by khaledhosny** [comment]
I guess this is because `unicodedata.script_extension()` returns `{'Phlp', 'Arab', 'Mand', 'Ougr', 'Syrc', 'Mani', 'Sogd', 'Adlm', 'Rohg'}` for 0x0640, but:
1. For characters that can b e used in multiple scripts, it should be added to all relevant features for these scripts, not only one of them.
2. If the font does not have any characters of a given script other than shared ones, it should not be considered (alternatively font’s `languagesystem`s should to be consulted).

**Pull request merged**